src/battle_script_commands.c

```diff
@@ -2243,6 +2243,11 @@ static void Cmd_healthbarupdate(void)
             if (GetBattlerSide(battler) == B_SIDE_PLAYER && gBattleMoveDamage > 0)
                 gBattleResults.playerMonWasDamaged = TRUE;
         }
+        if (!(gHitMarker & HITMARKER_PASSIVE_DAMAGE) && !DoesSubstituteBlockMove(gBattlerAttacker, battler, gCurrentMove)
+            && gBattleStruct->timesGotHit[gBattlerPartyIndexes[battler]][GetBattlerSide(battler)] < 6)
+        {
+            gBattleStruct->timesGotHit[gBattlerPartyIndexes[battler]][GetBattlerSide(battler)]++;
```

Comment:
I recommend not putting this here and instead adding it to the same place the logic for `MOVE_RAGE` is handled, considering that Rage Fist is just a buffed version of Rage.